dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/CRLFValidationHandler.java

```diff
@@ -14,55 +14,44 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelHandler.Sharable;
 
-import io.netty.util.Attribute;
-import io.netty.util.AttributeKey;
-
+/**
+ * Utility handler to validate that a request is not preceeded by more than two consecutive CRLF characters. 
+ */
+@Sharable
 public class CRLFValidationHandler extends ChannelInboundHandlerAdapter {
 
     private static final int MAX_CRLF_ALLOWED = 2;
+    public static final CRLFValidationHandler INSTANCE = new CRLFValidationHandler();
+
+    private CRLFValidationHandler() {}
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
-        if(msg instanceof ByteBuf){
-            ByteBuf buffer = (ByteBuf) msg;
-            buffer.markReaderIndex();
+        if(!(msg instanceof ByteBuf)){
+            ctx.fireChannelRead(msg);
+            return;
+        }
 
-            int leadingCRFLCount = 0;
-            boolean nonCRLFFound = false;
-            byte b;
-            byte nextByte;
+        ByteBuf buffer = (ByteBuf) msg;
+        buffer.markReaderIndex();
 
-            while(buffer.isReadable() && !nonCRLFFound){
-                b = buffer.readByte();
+        int leadingCRLFCount = 0;
+        while (buffer.isReadable()) {
+            byte b = buffer.readByte();
+            if (b != '\r')
+                break;
+            if (!buffer.isReadable() || buffer.readByte() != '\n')
+                break;
 
-                if(b == '\r'){
-                    if(buffer.isReadable()){
-                        nextByte = buffer.readByte();
-                        if(nextByte == '\n'){
-                            leadingCRFLCount++;
-                            if (leadingCRFLCount > MAX_CRLF_ALLOWED){
-                                ctx.channel().attr(NettyHttpConstants.THROW_FFDC).set(true);
-                                throw new IllegalArgumentException("Too many leading CRLF characters");
-                            }
-                        } else {
-                            nonCRLFFound = true;
-                            buffer.readerIndex(buffer.readerIndex() -1);
-                        }
-                    } else{
-                        nonCRLFFound = true;
-                    }
-                } else{
-                    nonCRLFFound = true;
-                }
+            if (++leadingCRLFCount > MAX_CRLF_ALLOWED) {
+                ctx.channel().attr(NettyHttpConstants.THROW_FFDC).set(true);
+                throw new IllegalArgumentException("Too many leading CRLF characters");
```

Comment:
Add to translation issue for review

Reply:
We use a similar message [here](https://github.com/OpenLiberty/open-liberty/blob/c8249ca16f2cb8e01cf5b098402ebedd7e7817f7/dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/GenericMessageImpl.java#L238) in Legacy. Should we add it or create a separate issue?